core/input/input.cpp

```diff
@@ -562,9 +562,9 @@ void Input::_parse_input_event_impl(const Ref<InputEvent> &p_event, bool p_is_em
 				button_event->set_pressed(st->is_pressed());
 				button_event->set_button_index(MOUSE_BUTTON_LEFT);
 				if (st->is_pressed()) {
-					button_event->set_button_mask(mouse_button_mask | (1 << (MOUSE_BUTTON_LEFT - 1)));
+					button_event->set_button_mask(MouseButton(mouse_button_mask | MOUSE_BUTTON_MASK_LEFT));
```

Comment:
I'm not an expert on C++ enums, are we sure that this is equivalent to casting?

Reply:
Yes, here is some minimal test code:

```cpp
enum FooBar {
	FOOBAR_FOO = 1,
	FOOBAR_BAR = 2,
};

FooBar foobar = FooBar(FOOBAR_FOO | FOOBAR_BAR);

print_line("FooBar foobar is: " + itos(foobar));
```

This outputs `FooBar foobar is: 3`. C++ enums are integers under the hood (as in most languages).

Note that in a comparison with an integer that is not in the Key enum, C++ knows that's often wrong, so an error will be thrown, so `foobar == 3` will give an error that it's always false (except that if you cast it back to an integer first, it won't always be false).